src/playground/sample_data_ingestion/sample_data_ingestor/sample_data_ingestor.py

```diff
@@ -0,0 +1,345 @@
+        
+            # Update timestamp with current time
+            entry[timestamp] = milliseconds
+            dataset.append(dumps(entry))
+        # Increment current date
+        current_date += timedelta(minutes = minutes)
+    return dataset
+
+
+def ingest_from_user_data(filename):
+    """
+    Utility function that loads user provided data to a list
+
+    Arguments:
+        - filename: The name of the user provided data file as a .json or .csv file (zipped or unzipped)
+
+    Returns:
+        - A list of data loaded from the file
+
+    Raises:
+        TypeError: Filename is not a string or is not a valid filename
+    """
+
+    # Input validation
+    if type(filename) is not str or (type(filename) is str and "." not in filename):
+        raise TypeError("Filename is not a string or is not a valid filename")
+    name = filename.split(".gz")[0]
+    dataset = []
+
+    # Unzips file 
+    if ".gz" in name:
+        with gzip.open(name, 'rt') as fin:
+                with open(name.split(".gz")[0], 'wb') as fout:
+                    copyfileobj(fin, fout)
+    # Reads from CSV
+    if ".csv" in name:
+        with open(name, "r") as f:
+            reader = csv.reader(f)
+            fields = next(reader)
+            for row in reader:
+                entry = {}
+                for i in range(len(fields)):
+                    entry[fields[i]] = row[i]
+                dataset.append(entry)
+    # Reads from JSON 
+    elif ".json" in name:
+        with open(name, "r") as f:
+            for line in f:
+                dataset.append(loads(line))
+    
+    # Removes zipped file
+    if filename != name:
+        remove(name)
+
+    return dataset
+
+
+def ingest_validation(template, index_name, client, avg_min, avg_max, abs_min, abs_max, anomaly_percentage):
+    """
+    Function that raises errors for improper arguments
+
+    Arguments:
+        - client: an OpenSearch Python client object
+        - template: JSON file, CSV file, JSON mapping, or JSON short-hand template
+        - index_name: The name of the index to ingest data
+        - avg_min: The lowest average value in which there is no anomaly
+        - avg_max: The highest average value in which there is no anomaly
+        - abs_min: The lowest possible value, which can be an anomaly
+        - abs_max: The highest possible value, which can be an anomaly
+        - anomaly_percentage: How often an anomaly occurs (minimum: 0.001)
+
+    Raises:
+        - ValueError: You must have a template and an index name to continue
+        - ValueError: Client needed to connect
+        - ValueError: Check ranges for generating trend data
+        - ValueError: Anomaly percentage cannot be outside a percent
+    """
+    if not template or not index_name:
+        raise ValueError("You must have a template and an index name to continue")
+    if not client:
```

Comment:
I'm confused here, do we validate something here? for example, if customer use client = "abc", then what's happening? thanks

Reply:
Done, i modified `ingest_validation`, which includes validation that `client` is an `OpenSearch` object